### DISCUSSION TITLE
How to change the geometries of an object.

### DISCUSSION THREAD

**Discussion by LeoCleme** [IfcOpenShell/IfcOpenShell#5411] [Q&A]
I need to change the geometries of an existing object or his representation. Like if i have a Pole 1 metre tall, how i can edit it to 1,5 metres tall.

**Comment by aothms** [comment]
This is a very general question. You can use Bonsai https://bonsaibim.org/ to have a UI to do so. https://bonsaibim.org/

If you want to use code you'd have to go in and edit the data. If you're lucky the element is represented as an IfcExtrudedAreaSolid and you can just edit the depth attribute of the extrusion. If you're less lucky the element is represented as a explicit faces and you might still be able to move the upper points' z coordinates (if they're not being reused). IFC can be a complex data model to make changes to.

**Reply by LeoCleme** [reply]
Ok, thank you for the answer!
